### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by AyomipoMajek**
OOP school library: add basic UI project done according to microverse requirement which include:

Watch the video with the UI example again (you have seen it while reading the Sneak Peek).
UI example
Your console app should behave in the same way as it is in the example.
Create a app.rb file that will serve as your console app entry-point. It should have methods that do the following:
List all books.
List all people.
Create a person (teacher or student, not a plain Person).
Create a book.
Create a rental.
List all rentals for a given person id.
In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:
Present the user with a list of options to perform.
Lets users choose an option.
If needed, ask for parameters for the option.
Have a way to quit the app.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+def display_menu
+  puts 'What would you like to do?'
+  puts '1. List all books'
+  puts '2. List all people'
+  puts '3. Create a person'
+  puts '4. Create a book'
+  puts '5. Create a rental'
+  puts '6. List all rentals for a given person id'
+  puts '0. Quit'
```
Comment (tugsem) [inline main.rb]: - [ ] Please add the welcoming title before options and update the last option to '7' as in the video, also when '7' is selected kindly return this message:
image


> image
>image

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  end
+end
+
+def create_person(app)
+  puts "Enter the person's name:"
+  name = gets.chomp
+  puts "Enter the person's age:"
+  age = gets.chomp.to_i
+  puts "Enter the person's type (student or teacher):"
+  type = gets.chomp.downcase
```
Comment (tugsem) [inline main.rb]: - [ ] In the option 3, we should see the below prompt, kindly fix this: 
>
image
> your app: 
image

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+end
+
+def create_book(app)
+  puts 'Enter the book title:'
+  title = gets.chomp
+  puts 'Enter the book author:'
+  author = gets.chomp
+  puts 'Enter the book isbn:'
+  isbn = gets.chomp
+  app.create_book(title, author, isbn)
```
Comment (tugsem) [inline main.rb]: - [ ] Please remove 'isbn' as we only need the title and the author from the user to stick with the below image from the video presentation, print the inputs on the same line and return success message after created a book.
image
image

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @AyomipoMajek 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️linters are passing
✔️correct use of gitflow
✔️professional documentation
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (AyomipoMajek) [reply main.rb]: Done, thank you

File: main.rb (hunk 2 above)
Comment (AyomipoMajek) [reply main.rb]: Done, thank you

File: main.rb (hunk 3 above)
Comment (AyomipoMajek) [reply main.rb]: Done, thank you

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @AyomipoMajek 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Gitflow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally

Required Changes ♻️
Check the comments under the review
- [ Required] Please kindly check if the changes requested by the previous reviewer have been implemented and pushed remotely. i don't seem to see the changes.  you are almost done

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: main.rb [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    puts 'Student created successfully'
+  elsif type == '2'
+    puts "Enter the teacher's specialty:"
+    specialty = gets.chomp
+    app.create_teacher(name, age, specialty)
+    puts 'Teacher created successfully'
+  else
+    puts 'Invalid person type'
+  end
+end
```
Comment (rotimiazeez) [inline main.rb]: Please note when I try to create a person (Student/Teacher), I'm met with an error. Kindly fix this 👍🏽
Screenshot from 2023-04-14 18-39-05

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
Hi @AyomipoMajek 😉, 

Great work on making the changes 👏 
You've implemented some of the requested changes well, but some still need to be addressed.

## Highlights 
- Good communication with the CR ✔️

## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

**Review by Ismailco — APPROVED** [review]
### Hi @AyomipoMajek, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ Nice UI
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**